src/Parser.zig

```diff
@@ -514,6 +521,42 @@ fn pragma(p: *Parser) Compilation.Error!bool {
     return found_pragma;
 }
 
+/// Issue errors for top-level definitions whose type was never completed.
+fn diagnoseIncompleteDefinitions(p: *Parser) !void {
+    @setCold(true);
+
+    const node_slices = p.nodes.slice();
+    const tags = node_slices.items(.tag);
+
+    var it = p.tentative_defs.iterator();
+    while (it.next()) |item| {
```

Comment:
Could this be made more efficient by looping over the decls and checking if there is an entry in the hash map?

Reply:
Definitely...that way we only have to loop over the decls once instead of N times